lib/Mojo/Redis.pm

```diff
@@ -32,7 +32,23 @@ has pubsub => sub {
 has url => sub { Mojo::URL->new($ENV{MOJO_REDIS_URL}) };
 
 # TODO: Should this attribute be public?
-has _blocking_connection => sub { shift->_connection(ioloop => Mojo::IOLoop->new) };
+sub _blocking_connection {
+  my $self = shift;
+
+  delete @$self{qw(blocking_connection pid queue)} unless ($self->{pid} //= $$) eq $$;    # Fork-safety
+
+  if (@_) {
+    $self->{blocking_connection} = shift;
+    return $self;
+  }
+
+  # Existing connection
+  return $self->{blocking_connection}->encoding($self->encoding)
+    if $self->{blocking_connection} and $self->{blocking_connection}->is_connected;
```

Comment:
Why do you check `is_connected()`?

Reply:
Because _dequeue does, so that it reconnects if it's been disconnected since you last retrieved a db object. But it might need more thought if the connection is not stored in the db object, since you don't want to reconnect except for the first time a db object needs the connection.